WordPress/src/main/java/org/wordpress/android/viewmodel/pages/PageParentViewModel.kt

```diff
@@ -128,4 +152,74 @@ class PageParentViewModel
         }
         return children + grandchildren
     }
+
+    fun onSearchExpanded(restorePreviousSearch: Boolean) {
+        if (isSearchExpanded.value != true) {
+            if (!restorePreviousSearch) {
+                clearSearch()
+            }
+            _isSearchExpanded.value = true
+        }
+    }
+
+    fun onSearchCollapsed() {
+        _isSearchExpanded.value = false
+        clearSearch()
+
+        launch {
+            delay(SEARCH_COLLAPSE_DELAY)
+        }
+    }
+
+    private fun clearSearch() {
+        _lastSearchQuery = ""
+        _searchPages.postValue(null)
+        _isSaveButtonVisible.postOnUi(false)
+        _currentParent.isSelected = false
+    }
+
+    fun onSearch(searchQuery: String, delay: Long = SEARCH_DELAY) {
+        searchJob?.cancel()
+        if (searchQuery.isNotEmpty()) {
+            searchJob = launch {
+                delay(delay)
+                searchJob = null
+                if (isActive) {
+                    _lastSearchQuery = searchQuery
+                    val result = search(searchQuery)
+                    _searchPages.postValue(result)
+                }
+            }
+        } else {
+            clearSearch()
+        }
+    }
+
+    private suspend fun search(
+        searchQuery: String
+    ): List<PageItem> = withContext(defaultDispatcher) {
+        val tempPages = _pages.value
+        if (tempPages != null) {
+            if (tempPages.isNotEmpty()) {
+                return@withContext tempPages
+                        .filterIsInstance(ParentPage::class.java)
```

Comment:
AFAIK unless there is a bug, this shouldn't filter out anything, right? If that's the case we might want to consider using `.also` and logging an exception if the list contains a non-parent page. The current implementation might just hide a bug which would be more difficult to find, wdyt? 

SideNote: Same thing probably applies to `it.id != 0L`.

Reply:
Actually I have that filter in there to get rid of any "Divider" lines which are not ParentPage types. The Top Level Parent item is a ParentPage, but can be identified by id of "0". I had made the assumption that neither of those items should be searchable, so I exclude them. So it is possible for the items to contain non ParentPages and that doesn't mean it's a bug.

Reply:
ohh, I see. Thanks for the explanation!